raiden/network/proxies/token_network.py

```diff
@@ -914,11 +914,14 @@ def settle(
             receipt_or_none = check_transaction_threw(self.client, transaction_hash)
             if receipt_or_none:
                 log.info('settle failed', **log_details)
-                self._check_channel_state_for_settle(
-                    self.node_address,
-                    partner,
-                    channel_identifier,
-                )
+                try:
+                    self._check_channel_state_for_settle(
+                        self.node_address,
+                        partner,
+                        channel_identifier,
+                    )
+                except RaidenRecoverableError:
```

Comment:
The error message in the issue doesn't seems like a recoverable error:

> raiden.exceptions.RaidenRecoverableError: Channel cannot be settled before closing

@rakanalh any ideas why this is recoverable?

Reply:
@rakanalh ping

Reply:
I think the message is wrong. 
https://github.com/raiden-network/raiden/pull/2168/files#diff-d02574a7a7eca826412780e80a6326a3R1190

The function is called if the settle transaction failed to figure out why the failure happened.
Since the channel is closed, but the current block exceeded the settlement block then it could be that the channel was already settled where both participants were racing to settle and the partner won that race.

Reply:
This doesn't make sense, if the channel was settled then we would have the [recoverable error above](https://github.com/raiden-network/raiden/pull/2168/files#diff-d02574a7a7eca826412780e80a6326a3R1171).

I think the fix introduced by the PR is fine, but I'm not convinced the exception raised is recoverable. It looks to me that we have a bug in there.

Reply:
What do you mean? The exception raised is as described [here](https://github.com/raiden-network/raiden/pull/2432#issuecomment-420652774).

In such a scenario the channel is just unusable anymore, but raiden can still go on, so it's a recoverable exception.